code/modules/mob/living/living.dm

```diff
@@ -1429,6 +1419,34 @@
 		log_mapping(ERROR_ERROR_LANDMARK_ERROR)
 		CRASH(ERROR_ERROR_LANDMARK_ERROR)
 
+/**
+  * Changes the inclination angle of a mob, used by humans and others to differentiate between standing up and prone positions.
+  *
+  * In BYOND-angles 0 is NORTH, 90 is EAST, 180 is SOUTH and 270 is WEST.
+  * This usually means that 0 is standing up, 90 and 270 are horizontal positions to right and left respectively, and 180 is upside-down.
+  * Mobs that do now follow these conventions due to unusual sprites should require a special handling or redefinition of this proc, due to the density and layer changes.
+  * The return of this proc is the previous value of the modified lying_angle if a change was successful (might include zero), or null if no change was made.
+  */
+/mob/living/proc/set_lying_angle(new_lying)
+	if(new_lying == lying_angle)
+		return
+	. = lying_angle
+	lying_angle = new_lying
+	if(lying_angle != lying_prev)
+		update_transform()
+		lying_prev = lying_angle
+	if(lying_angle != 0)
+		if(layer == initial(layer)) //to avoid things like hiding larvas.
+			layer = LYING_MOB_LAYER //so mob lying always appear behind standing mobs
+		if(. != 0)
+			density = FALSE
```

Comment:
not following the usage of `.` here. So it stores the old lying_angle before it's set, but then is only used at the comparison at 1441? But that's the pre-set angle so that wouldn't work right, not sure if there's something I'm missing here.

Reply:
`.` merely stores the previous value `lying_angle`
The usage for it is mostly for when you redefine the proc, you can run `. = ..()` and have both the new and old `lying_angle` values, or `null` in `.` if it is unchanged and needs not running.

Reply:
Ah, it's for overrides, that's understandable. Still not following the density change logic though

Reply:
Commented it to explain better.
In short, if you were standing up before and now you are prone, you density gets set to `FALSE`.
If you were prone and you get set to no longer (`lying = 0`) then you re-gain density.
This happened on the `update_mobility()` proc, it just got moved here to only happen when `lying` makes a meaningful change.
That way if we want for things to react to density changes we can add a proc/signal that will only be invoked at the right time, for example.

Reply:
Makes sense.